packages/flutter/lib/src/animation/scroll_behavior.dart

```diff
@@ -105,26 +105,25 @@ class UnboundedBehavior extends ExtentScrollBehavior {
 Simulation _createFlingScrollSimulation(double position, double velocity, double minScrollOffset, double maxScrollOffset) {
   double startVelocity = velocity * _kSecondsPerMillisecond;
 
-  // Assume that we're rendering at atleast 15 FPS. Stop when we're
-  // scrolling less than one logical pixel per frame. We're essentially
-  // normalizing by the devicePixelRatio so that the threshold has the
-  // same effect independent of the device's pixel density.
-  double endVelocity = 15.0 * ui.window.devicePixelRatio;
+  // Assume that we're rendering at >= 15 FPS. Stop when we're
+  // scrolling less than one physical pixel per frame, which isn't
+  // visually detectable.
+  double endVelocity = 15.0 / ui.window.devicePixelRatio;
```

Comment:
The problem with choosing a large value like 500ms is that we don't fire the `onScrollEnd` callback until we finish the animations, which means there's a cost to coasting down to a tiny velocity.